src/containers/BooksForm.js

```diff
@@ -1,17 +1,60 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { createBook } from '../actions';
 
-const BooksForm = () => {
-  const options = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi']
-    .map(x => <option key={x} value={x}>{x}</option>);
-  return (
-    <div>
-      <input />
-      <select>
-        {options}
-      </select>
-      <button type="submit">Submit</button>
-    </div>
-  );
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: 'Action',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const { id, value } = event.target;
+    this.setState({ [id]: value });
+  }
+
+  handleSubmit() {
+    const { title, category } = this.state;
+    const { addBook } = this.props;
+    if (title) {
+      addBook({ id: `_${Math.random().toString(36).substr(2, 9)}`, title, category });
+      this.setState({ title: '', category: 'Action' });
+    }
+    return true;
+  }
+
+  render() {
+    const options = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi']
+      .map(x => <option key={x} value={x}>{x}</option>);
+    const { title, category } = this.state;
+    return (
+      <div>
+        <input id="title" onChange={this.handleChange} value={title} />
+        <select id="category" onChange={this.handleChange} value={category}>
+          {options}
+        </select>
+        <button type="submit" onClick={this.handleSubmit}>Submit</button>
+      </div>
```

Comment:
We could add native HTML form validation. By using the `form` element with `onSubmit` event handler. This will need the `title` `input` element to also include the `required` attribute.

Reply:
Thanks! I will change it